Lib/asyncio/staggered.py

```diff
@@ -116,8 +119,10 @@ async def run_one_coro(previous_failed) -> None:
                 if i != this_index:
                     t.cancel()
 
-    first_task = loop.create_task(run_one_coro(None))
+    ok_to_start = locks.Event()
+    first_task = loop.create_task(run_one_coro(ok_to_start, None))
     running_tasks.append(first_task)
```

Comment:
Actually, while we're here, I'd replace `running_tasks` with an explicit `TaskGroup` (which should be passed to `run_one_coro` via an argument for other nested tasks.

Reply:
This would also solve this problem: https://github.com/python/cpython/pull/124847#issuecomment-2393992545

Reply:
We tried to use a TaskGroup before, but it doesn't abort fast enough to avoid two tasks winning at the same time. This is resolved in trio by closing socks that are immediate runner ups